src/gigs-board/pages/Create.jsx

```diff
@@ -0,0 +1,537 @@
+  widgetProps = {
+    ...widgetProps,
+    nearDevGovGigsContractAccountId: props.nearDevGovGigsContractAccountId,
+    nearDevGovGigsWidgetsAccountId: props.nearDevGovGigsWidgetsAccountId,
+    referral: props.referral,
+  };
+  return (
+    <Widget
+      src={`${nearDevGovGigsWidgetsAccountId}/widget/gigs-board.${widgetName}`}
+      props={widgetProps}
+      key={key}
+    />
+  );
+}
+
+function href(widgetName, linkProps) {
+  linkProps = { ...linkProps };
+  if (props.nearDevGovGigsContractAccountId) {
+    linkProps.nearDevGovGigsContractAccountId =
+      props.nearDevGovGigsContractAccountId;
+  }
+  if (props.nearDevGovGigsWidgetsAccountId) {
+    linkProps.nearDevGovGigsWidgetsAccountId =
+      props.nearDevGovGigsWidgetsAccountId;
+  }
+  if (props.referral) {
+    linkProps.referral = props.referral;
+  }
+  const linkPropsQuery = Object.entries(linkProps)
+    .map(([key, value]) => `${key}=${value}`)
+    .join("&");
+  return `#/${nearDevGovGigsWidgetsAccountId}/widget/gigs-board.pages.${widgetName}${
+    linkPropsQuery ? "?" : ""
+  }${linkPropsQuery}`;
+}
+/* END_INCLUDE: "common.jsx" */
+
+const parentId = props.parentId ?? null;
+const postId = props.postId ?? null;
+const mode = props.mode ?? "Create";
+
+const referralLabels = props.referral ? [`referral:${props.referral}`] : [];
+const labelStrings = (props.labels ?? []).concat(referralLabels);
+const labels = labelStrings.map((s) => {
+  return { name: s };
+});
+
+initState({
+  seekingFunding: false,
+  //
+  author_id: context.accountId,
+  // Should be a list of objects with field "name".
+  labels,
+  // Should be a list of labels as strings.
+  // Both of the label structures should be modified together.
+  labelStrings,
+  postType: "Idea",
+  name: props.name ?? "",
+  description: props.description ?? "",
+  amount: props.amount ?? "",
+  token: props.token ?? "Near",
+  supervisor: props.supervisor ?? "neardevgov.near",
+  githubLink: props.githubLink ?? "",
+  warning: "",
+});
+
+// This must be outside onClick, because Near.view returns null at first, and when the view call finished, it returns true/false.
+// If checking this inside onClick, it will give `null` and we cannot tell the result is true or false.
+let grantNotify = Near.view("social.near", "is_write_permission_granted", {
+  predecessor_id: nearDevGovGigsContractAccountId,
+  key: context.accountId + "/index/notify",
+});
+if (grantNotify === null) {
+  return;
+}
+
+const onSubmit = () => {
+  let labels = state.labelStrings;
+
+  let body = {
+    name: state.name,
+    description: generateDescription(
+      state.description,
+      state.amount,
+      state.token,
+      state.supervisor
+    ),
+  };
+
+  if (state.seekingFunding && state.postType === "Solution") {
+    body = {
+      ...body,
+      post_type: "Submission",
+    };
+  } else if (state.postType === "Solution") {
+    // Solution
+    body = {
+      ...body,
+      post_type: "Submission",
+    };
```

Comment:
There is now unnecessary duplication, and `submission_version` is required:

```suggestion
  if (state.postType === "Solution") {
    body = {
      ...body,
      post_type: "Submission",
      submission_version: "V1",
    };
```